### ISSUE TITLE
[Bug]: `needs` and `steps` outputs context checking should be case sensitive

### ISSUE THREAD

**Issue by Nokel81** [rhysd/actionlint#271]
If I have the following workflow in my repo as `.github/workflows/my-workflow.yml`

```yml
name: My name
on:
  workflow_dispatch:
jobs:
  tag:
    outputs:
      tagname: ...
    runs-on: ubuntu-latest
    steps:
      - name: My step
        run: echo Foo
  other:
    needs: [tag]
    runs-on: ubuntu-latest
    steps:
      - name: Second
        run: echo ${{ needs.tag.outputs.TAGNAME }} # this lint
```

and I run `actionlint` at the root of the repo I do not see any lint errors. But if I change `# this lint` to be ` run: echo ${{ needs.tag.outputs.TAGNAME1 }}` then I do see a lint error

**Comment by rhysd** [comment]
I'm not seeing any problem here.

I tried your workflow and it echoed `...` at `echo ${{ needs.tag.outputs.TAGNAME }}` correctly. So it looks correct that actionlint reports no lint errors.

https://github.com/rhysd/misc/actions/runs/5560379843/jobs/10157337157

> But if I change # this lint to be  run: echo ${{ needs.tag.outputs.TAGNAME1 }} then I do see a lint error

That's because `TAGNAME1` (or `tagname1`) doesn't exist in outputs of `tag` job. It looks also correct.

**Comment by rhysd** [comment]
OK, then I think we can close this ticket.
